app/models/assignment.rb

```diff
@@ -11,8 +11,10 @@ class Assignment < ApplicationRecord
                                 foreign_key: :assignment_uuid,
                                 dependent: :destroy,
                                 inverse_of: :assignment
+
+  as = arel_table
   has_one :exercise_calculation,
-    -> { where '"exercise_calculations"."ecosystem_uuid" = "assignments"."ecosystem_uuid"' },
+    -> { where arel_table[:ecosystem_uuid].eq(as[:ecosystem_uuid]) },
     primary_key: :student_uuid,
```

Comment:
I'm confused by this.  Why assign `arel_table` to `as`?
I'm guessing it's because inside the block, `arel_table ` is referring to `exercise_calculation`?  If so, I think it'd be clearer to be explicit
```ruby
-> { where ExerciseCalculation.arel_table[:ecosystem_uuid].eq(Assignment.arel_table[:ecosystem_uuid]) },
```

Reply:
Yes... I'll change it